src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -524,41 +524,69 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
     }
   };
 
-  onCopy = async (
+  onDuplicate = async (
     savedObjects: SavedObjectWithMetadata[],
     includeReferencesDeep: boolean,
     targetWorkspace: string
   ) => {
     const { notifications, http } = this.props;
-    const objectsToCopy = savedObjects.map((obj) => ({ id: obj.id, type: obj.type }));
+    const objectsToDuplicate = savedObjects.map((obj) => ({ id: obj.id, type: obj.type }));
     let result;
     try {
-      result = await copySavedObjects(http, objectsToCopy, includeReferencesDeep, targetWorkspace);
+      result = await duplicateSavedObjects(
+        http,
+        objectsToDuplicate,
+        includeReferencesDeep,
+        targetWorkspace
+      );
       if (result.success) {
         notifications.toasts.addSuccess({
-          title: i18n.translate('savedObjectsManagement.objectsTable.copy.successNotification', {
-            defaultMessage:
-              'Copy ' + savedObjects.length.toString() + ' saved objects successfully',
-          }),
+          title: i18n.translate(
+            'savedObjectsManagement.objectsTable.duplicate.successNotification',
+            {
+              defaultMessage:
+                'Duplicate ' + savedObjects.length.toString() + ' saved objects successfully',
+            }
+          ),
         });
       } else {
-        const failedCount = savedObjects.length - result.successCount;
-        notifications.toasts.addSuccess({
-          title: i18n.translate('savedObjectsManagement.objectsTable.copy.dangerNotification', {
-            defaultMessage: 'Unable to copy ' + failedCount.toString() + ' saved objects',
-          }),
+        let failedCount = savedObjects.length;
+        if (result.successCount) {
+          // reduplicate objects which can be successfully duplicated
+          const successObjectIds = result.successResults.map((item: { id: string }) => item.id);
+          const successObjectsToDuplicate = objectsToDuplicate.filter((item) =>
+            successObjectIds.includes(item.id)
+          );
+          result = await duplicateSavedObjects(
+            http,
+            successObjectsToDuplicate,
+            includeReferencesDeep,
+            targetWorkspace
+          );
+          if (result.success) {
+            failedCount -= successObjectsToDuplicate.length;
+          }
+        }
```

Comment:
I don't understand this logic, seems it duplicates the successful objects when duplicate failed?

Reply:
When duplicating several objects, it is possible that some of the saved objects can be successfully duplicated while others cannot. For example, if we try to duplicate 5 saved objects and 1 get failed, all these five saved objects will not be duplicated. These objects which can be duplicated is returned in `result.successResults`. We need this retry duplicating so that these objects can be duplicated.